### PR TITLE
Setup cmake

### PR DESCRIPTION
**Pull request by SourenP**

### CODE REVIEW

File: src/client/systems/NetworkingClient.cpp [hunk 1]
```diff
@@ -15,150 +15,218 @@
...
+int NetworkingClient::getPlayerId() const {
+    return mPlayerId_ta;
+}
 
-    if (!mIsRegistered) {
-        throw std::runtime_error("Failed to register. Exiting.");
-    }
+GameState NetworkingClient::getGameState() {
+     std::lock_guard<std::mutex> mGameState_guard(mGameState_lock);
+    mGameStateIsFresh_ta = false;
```
Comment (copacetic) [inline src/client/systems/NetworkingClient.cpp]: what is _ta?

**Comment by copacetic** [comment]
can you rebase when ready? definitely want to see a description of why all the work with mutexes happened to protect the thread state when shutting down

File: src/client/systems/NetworkingClient.cpp (hunk 1 above)
Comment (SourenP) [reply src/client/systems/NetworkingClient.cpp]: I added a description in the README.txt about this.

**Comment by SourenP** [comment]
> can you rebase when ready? definitely want to see a description of why all the work with mutexes happened to protect the thread state when shutting down

`518da75` Rewrite NetworkingClient and NetworkingServer to use explicit mutex locks+guards instead of using custom written thread safe classes.

**Pull request merged**